cfme/utils/appliance/__init__.py

```diff
@@ -2760,9 +2796,102 @@ class BaseEntity(NavigatableMixin):
     argument names have been used.
     """
 
-    def __new__(cls, *args, **kwargs):
-        # DISABLED as breaking 'copy' operations
-        # first_arg = args[0] if args else kwargs.get('collection')
-        # if not first_arg or not isinstance(first_arg, BaseCollection):
-        #     raise Exception('First argument must be a collection')
-        return super(BaseEntity, cls).__new__(cls)
+    def __init__(self, collection, *args, **kwargs):
+        self.collection = collection
+        self.appliance = self.collection.appliance
+        self.init(*args, **kwargs)
+
+    def init(self, *args, **kwargs):
+        pass
+
+    @property
+    def collections(self):
+        if not self._collections_obj:
+            self._collections_obj = ObjectCollections(self)
+        return self._collections_obj
+
+
+@attr.s
+class BaseCollection2(NavigatableMixin):
+    """Class for helping create consistent Collections
+
+    The BaseCollection class is responsible for ensuring two things:
+
+    1) That the API consistently has the first argument passed to it
+    2) That that first argument is an appliance instance
+
+    This class works in tandem with the entrypoint loader which ensures that the correct
+    argument names have been used.
+    """
+
+    ENTITY = None
+
+    parent = attr.ib()
+    filters = attr.ib(default=attr.Factory(dict))
```

Comment:
initially i thougth i would dislike filters as collection attribute on construction

however strategically its more beneficial to go from collection to entity with collection in such cases so its fine 👍